api/api/views/media_views.py

```diff
@@ -272,7 +272,7 @@ def related(self, request, identifier=None, *_, **__):
             raise APIException(getattr(e, "message", str(e)))
         # If there are no hits in the search controller
         except IndexError:
-            raise APIException("Could not find items.", 404)
```

Comment:
If this was supposed to be raising a 404, why wasn't it working before?

Reply:
Because the second parameter isn't the status code, it's the _error_ code: https://github.com/encode/django-rest-framework/blob/master/rest_framework/exceptions.py#L99-L114. Typically this is something like a slug that identifies the error type within the status code: https://github.com/encode/django-rest-framework/blob/master/rest_framework/exceptions.py#L183-L192. This lets you have multiple distinct error types within a single semantic HTTP status code that are identifiable on the client without needing to parse the potentially arbitrary error message string.